pkg/river/parser/internal.go

```diff
@@ -276,10 +276,12 @@ func (p *parser) parseBlockName() *blockName {
 
 	// [ string ]
 	if p.tok != token.ASSIGN && p.tok != token.LCURLY {
-		if p.tok == token.STRING && len(p.lit) > 2 {
-			bn.Label = p.lit[1 : len(p.lit)-1] // Strip quotes from label
-			if !isValidIdentifier(bn.Label) {
-				p.addErrorf("expected block label to be a valid identifier")
```

Comment:
This was a small unexpected parse error: `a.b.c ""` was returning an error at the parse level, while I was expecting to see an issue at the evaluator level.